src/Engine/EmitMSIL/Internals.cs

```diff
@@ -0,0 +1,233 @@
+
+            // In order to follow the conventions of Java, Python, Scala and Google's calculator,
+            // the returned modulo will follow the sign of the divisor (not the dividend). 
+
+            if (intMod < 0 && b > 0 || intMod > 0 && b < 0)
+            {
+                intMod += b;
+            }
+
+            return intMod;
+        }
+
+        public static dynamic Neg(dynamic a)
+        {
+            return -a;
+        }
+
+        public static bool and(dynamic a, dynamic b)
+        {
+            bool lhs = ToBoolean(a);
+            bool rhs = ToBoolean(b);
+
+            return lhs && rhs;
+        }
+
+        public static bool or(dynamic a, dynamic b)
+        {
+            bool lhs = ToBoolean(a);
+            bool rhs = ToBoolean(b);
+
+            return lhs || rhs;
+        }
+
+        public static bool Not(dynamic a)
+        {
+            bool data = ToBoolean(a);
+
+            return !data;
+        }
+
+        public static bool eq(dynamic a, dynamic b)
+        {
+            if (a is bool || b is bool)
+            {
+                bool ab = ToBoolean(a);
+                bool bb = ToBoolean(b);
+                return ab == bb;
+            }
+
+            if (a is double || b is double || a is long || b is long)
+            {
+                if (a is double || b is double)
+                {
+                    double ad = a;
+                    double bd = b;
+                    return MathUtils.Equals(ad, bd);
+                }
+
+                return a == b;
+            }
+
+            if (a is string astr && b is string bstr)
+            {
+                return String.Compare(astr, bstr, StringComparison.InvariantCulture) == 0;
+            }
+
+            return a.Equals(b);
+        }
+
+        public static bool nq(dynamic a, dynamic b)
+        {
+            if (a is bool || b is bool)
+            {
+                bool ab = ToBoolean(a);
+                bool bb = ToBoolean(b);
+                return ab != bb;
+            }
+
+            if (a is double || b is double || a is long || b is long)
+            {
+                if (a is double || b is double)
+                {
+                    double ad = a;
+                    double bd = b;
+                    return !MathUtils.Equals(ad, bd);
+                }
+
+                return a != b;
+            }
+
+            if (a is string astr && b is string bstr)
+            {
+                return String.Compare(astr, bstr, StringComparison.InvariantCulture) != 0;
+            }
+
+            return !a.Equals(b);
+        }
+
+        public static bool gt(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
+            {
+                return a > b;
+            }
+
+            return false;
+        }
+
+        public static bool lt(dynamic a, dynamic b)
+        {
+            if (a is double || b is double || a is long || b is long)
```

Comment:
These functions assume all integer arguments to be `long`, which means we need to convert all returned `Int32`s to long. I actually don't think that's necessary anymore. We must convert them to `long` only when needed, which means that we should support all these methods for int32 as well.

Reply:
Thanks. This has confused me. I have never seen anything than `long` so far so thought that was on purpose.

Reply:
I needed to keep it long when emitting integer nodes to maintain the same behaviour with the old DS engine specs.

Reply:
Ok, so there is no need for me to support `int32` in these methods?